packages/react-dom/src/__tests__/ReactComponentLifeCycle-test.js

```diff
@@ -197,6 +197,32 @@ describe('ReactComponentLifeCycle', () => {
     }).not.toThrow();
   });
 
+  it("warns if setting 'this.state = props'", () => {
+    spyOn(console, 'error');
+
+    class StatefulComponent extends React.Component {
+      constructor(props, context) {
+        super(props, context);
+        this.state = props;
+      }
+      render() {
+        return <div />;
+      }
+    }
+
+    ReactTestUtils.renderIntoDocument(<StatefulComponent />);
+    expectDev(console.error.calls.count()).toBe(1);
+    expectDev(console.error.calls.argsFor(0)[0]).toContain(
+      'It looks like the StatefulComponent component contains a line like this in its constructor: ' +
+        'this.state = props; ' +
```

Comment:
Have you checked how this looks in a browser? Since you didn't add the linebreaks I assume it will look a bit confusing.

Reply:
Good point. Pushed a new commit with line breaks before and after the code excerpt to increase readability.